driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -420,8 +420,8 @@ public ConnectionString(final String connectionString, @Nullable final DnsClient
             + "'%s' contains the keys %s", ALLOWED_OPTIONS_IN_TXT_RECORD, unresolvedHosts.get(0), txtRecordsOptionsMap.keySet()));
         }
         Map<String, List<String>> combinedOptionsMaps = combineOptionsMaps(txtRecordsOptionsMap, connectionStringOptionsMap);
-        if (isSrvProtocol && !combinedOptionsMaps.containsKey("ssl")) {
-            combinedOptionsMaps.put("ssl", singletonList("true"));
+        if (isSrvProtocol && !(combinedOptionsMaps.containsKey("tls") || combinedOptionsMaps.containsKey("ssl"))) {
+            combinedOptionsMaps.put("tls", singletonList("true"));
```

Comment:
Switched to the non-deprecated parameter name

Reply:
Ack